src/components/Modal/base.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { cleanClasses } from './helpers';
+
+export interface Props {
```

Comment:
All these properties need to have clearly defined types.  Ex:
1. align can be an enum.  
2. children is not needed, react adds them.
3. col, colSmall, colMedium, colLarge make no sense.

If the property expects set number of values, like align, then it can be an enum.

@krunalm @ranmalr Can help you with these.